FAQ_FOR_MAC.md

```diff
@@ -0,0 +1,141 @@
+# FAQ for Mac Users
```

Comment:
Thanks for the PR.
This PR involves a re-write of the CMakeLists and might introduce new errors. 
It'd be good to look into how we can use the same file for compiling across gcc and clang, perhaps.